internal/motor/create_schema.go

```diff
@@ -0,0 +1,67 @@
+package motor
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/sonr-io/sonr/pkg/client"
+	"github.com/sonr-io/sonr/pkg/tx"
+	st "github.com/sonr-io/sonr/x/schema/types"
+	mt "go.buf.build/grpc/go/sonr-io/motor/api/v1"
+)
+
+func (mtr *MotorNode) CreateSchema(requestBytes []byte) (mt.CreateSchemaResponse, error) {
+	var request mt.CreateSchemaRequest
+	if err := json.Unmarshal(requestBytes, &request); err != nil {
+		return mt.CreateSchemaResponse{}, fmt.Errorf("unmarshal request: %s", err)
+	}
+
+	listFields, err := convertFields(request.Fields)
+	if err != nil {
+		return mt.CreateSchemaResponse{}, fmt.Errorf("process fields: %s", err)
+	}
+	createSchemaMsg := st.NewMsgCreateSchema(&st.SchemaDefinition{
+		Creator: mtr.Address,
+		Label:   request.Label,
+		Fields:  listFields,
+	})
+
+	txRaw, err := tx.SignTxWithWallet(mtr.Wallet, "/sonrio.sonr.schema.MsgCreateSchema", createSchemaMsg)
+	if err != nil {
+		return mt.CreateSchemaResponse{}, fmt.Errorf("sign tx with wallet: %s", err)
+	}
+
+	resp, err := mtr.Cosmos.BroadcastTx(txRaw)
+	if err != nil {
+		return mt.CreateSchemaResponse{}, fmt.Errorf("broadcast tx: %s", err)
+	}
+
+	csresp := &st.MsgCreateSchemaResponse{}
+	if err := client.DecodeTxResponseData(resp.TxResponse.Data, csresp); err != nil {
+		return mt.CreateSchemaResponse{}, fmt.Errorf("decode MsgCreateSchemaResponse: %s", err)
+	}
+
+	return mt.CreateSchemaResponse{
```

Comment:
we should return the full what is, as the schema module will require it.

Reply:
We might want to create a map of did -> whatIs, and query for the `SchemaDefinition` from ipfs once the schema is created. this could be implemented in a `retrieveSchemaDefinition` method to be explicit.

Reply:
Do you mean in the motor? I think a `GetSchema` method needs to exist in the chain.

As for keeping these cached in the motor, let's work out a design tomorrow (Wednesday) which allows us to cache schemas and update the cache.

Reply:
Added `WhatIs` to the response.